### PR TITLE
[MM-48963] - /cloud/products fails on Self-Hosted (Community)

### PR DESCRIPTION
**Pull request by AGMETEOR**
#### Summary
Prevent fetch of cloud products in deliquency modal when on self-hosted

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48963

#### Release Note

```release-note
NONE
```

### CODE REVIEW

**Review by yasserfaraazkhan — APPROVED** [review]

**Comment by neallred** [comment]
Thanks for reaching out! That error is expected at the moment because we merged the mattermost part of the self-hosted purchase code with the flag turned on, but the customer-web-server side isn't yet merged. No need to log a ticket yet. Given conversation in the customer-web-server PR (https://github.com/mattermost/customer-web-server/pull/1014#issuecomment-1353062185), it is likely we will merge the matching customer-web-server code and deploy it to the test environment and this error will not display after that.

File: components/delinquency_modal/useDelinquencyModalController.tsx [hunk 1]
```diff
@@ -45,7 +45,7 @@ export const useDelinquencyModalController = (props: UseDelinquencyModalControll
     };
 
     useEffect(() => {
-        if (delinquencyModalPreferencesConfirmed.length === 0 && product == null && !requestedProducts) {
+        if (delinquencyModalPreferencesConfirmed.length === 0 && product === undefined && !requestedProducts && isCloud) {
```
Comment (marianunez) [inline components/delinquency_modal/useDelinquencyModalController.tsx:48]: Was this mounted even before accessing the modal? Just want to make sure this solves the problem that was reported in community

**Review by marianunez — APPROVED** [review]

File: components/delinquency_modal/useDelinquencyModalController.tsx (hunk 1 above)
Comment (AGMETEOR) [reply components/delinquency_modal/useDelinquencyModalController.tsx:48]: Yes, the modal is mounted in root

**Pull request merged**